htmd/projections/metriccoordinate.py

```diff
@@ -71,26 +86,26 @@ def project(self, mol):
         mol = mol.copy()
         if self.pbc:
             mol.wrap(self._centersel)
-        mol.align(trajalnsel, refmol=self._refmol, refsel=self._refalnsel)
+
+        if trajalnsel is not None and self._refmol is not None:
+            mol.align(trajalnsel, refmol=self._refmol, refsel=self._refalnsel)
 
         coords = np.concatenate((mol.coords[atomsel, 0, :], mol.coords[atomsel, 1, :], mol.coords[atomsel, 2, :]))
         return np.transpose(coords)  # I need to permute the dimensions here to put frames on the first
 
     def _getSelections(self, mol):
-        if self._pc_trajalnsel is not None and self._pc_atomsel is not None and self._pc_centersel is not None:
```

Comment:
why treating all of them at once not like in metriccoords?

Reply:
I don't treat them all at once because you might have None alignment yet have an atomsel. I think the old way of doing a single if is ugly and I should probably refactor it in the old classes